examples/multimodal/utils/args.py

```
@@ -127,83 +116,60 @@ def base_parse_args(
-
-    for attempt in range(1, max_attempts + 1):
-        # Hold socket open just long enough to reserve in ETCD
-        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
-            sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-            sock.bind(("", 0))
-            port = sock.getsockname()[1]
-
-            # Reserve in ETCD while holding the socket
-            key = f"dyn://{namespace}/ports/{node_name}/{port}"
-            value = {
-                "worker_id": worker_id,
-                "reason": reason,
-                "reserved_at": time.time(),
-                "pid": os.getpid(),
-            }
-
-            try:
-                await etcd_client.kv_create(
-                    key=key,
-                    value=json.dumps(value).encode(),
-                    lease_id=etcd_client.primary_lease_id(),
-                )
-                logger.debug(f"Reserved OS-assigned port {port} for {worker_id}")
-                return port
-
-            except Exception as e:
-                logger.debug(
-                    f"Port {port} on {node_name} was already reserved (attempt {attempt}): {e}"
-                )
-
-        if attempt < max_attempts:
-            await asyncio.sleep(0.01)
-
-    raise RuntimeError(
-        f"Failed to allocate and reserve a port after {max_attempts} attempts"
+    context_json = {
+        "worker_id": worker_id,
+        "reason": reason,
+        "reserved_at": time.time(),
+        "pid": os.getpid(),
+        "block_size": 1,
+    }
+
+    # Any ephemeral port, equivalent to binding port 0
+    port_range_min = 32_768
+    port_range_max = 60_999
+    allocated_ports = await runtime.allocate_port_block(
+        namespace,
+        port_range_min,
+        port_range_max,
+        1,  # how many ports to allocate
+        json.dumps(context_json),
     )
+    if not allocated_ports:
+        raise RuntimeError("allocate_port_block returned no ports")
```

is there a case where it is OK for `allocate_port_block` to return `None`?

if not, let's just have it `raise` instead of returning `None`.

---

I see! Fixing, thanks.